php/php.y

```diff
@@ -0,0 +1,1165 @@
+%left '?' ':'
+%right T_COALESCE
+%left T_BOOLEAN_OR
+%left T_BOOLEAN_AND
+%left '|'
+%left '^'
+%left '&'
+%nonassoc T_IS_EQUAL T_IS_NOT_EQUAL T_IS_IDENTICAL T_IS_NOT_IDENTICAL T_SPACESHIP
+%nonassoc '<' T_IS_SMALLER_OR_EQUAL '>' T_IS_GREATER_OR_EQUAL
+%left T_SL T_SR
+%left '+' '-' '.'
+%left '*' '/' '%'
+%right '!'
+%nonassoc T_INSTANCEOF
+%right '~' T_INC T_DEC T_INT_CAST T_DOUBLE_CAST T_STRING_CAST T_ARRAY_CAST T_OBJECT_CAST T_BOOL_CAST T_UNSET_CAST '@'
+%right T_POW
+%right '['
+%nonassoc T_NEW T_CLONE
+// %left T_NOELSE
+%left T_ELSEIF
+%left T_ELSE
+%left T_ENDIF
+%right T_STATIC T_ABSTRACT T_FINAL T_PRIVATE T_PROTECTED T_PUBLIC
+
+// %token T_NOELSE
+%token T_LNUMBER
+%token T_DNUMBER
+%token T_STRING
+%token T_VARIABLE
+// %token T_INLINE_HTML
+// %token T_ENCAPSED_AND_WHITESPACE
+%token T_CONSTANT_ENCAPSED_STRING
+%token T_STRING_VARNAME
+%token T_NUM_STRING
+
+%token T_INCLUDE
+%token T_INCLUDE_ONCE
+%token T_EVAL
+%token T_REQUIRE
+%token T_REQUIRE_ONCE
+%token T_LOGICAL_OR
+%token T_LOGICAL_XOR
+%token T_LOGICAL_AND
+%token T_PRINT
+%token T_YIELD
+%token T_YIELD_FROM
+%token T_PLUS_EQUAL
+%token T_MINUS_EQUAL
+%token T_MUL_EQUAL
+%token T_DIV_EQUAL
+%token T_CONCAT_EQUAL
+%token T_MOD_EQUAL
+%token T_AND_EQUAL
+%token T_OR_EQUAL
+%token T_XOR_EQUAL
+%token T_SL_EQUAL
+%token T_SR_EQUAL
+%token T_BOOLEAN_OR
+%token T_BOOLEAN_AND
+%token T_IS_EQUAL
+%token T_IS_NOT_EQUAL
+%token T_IS_IDENTICAL
+%token T_IS_NOT_IDENTICAL
+%token T_IS_SMALLER_OR_EQUAL
+%token T_IS_GREATER_OR_EQUAL
+%token T_SPACESHIP
+%token T_SL
+%token T_SR
+%token T_INSTANCEOF
+%token T_INC
+%token T_DEC
+%token T_INT_CAST
+%token T_DOUBLE_CAST
+%token T_STRING_CAST
+%token T_ARRAY_CAST
+%token T_OBJECT_CAST
+%token T_BOOL_CAST
+%token T_UNSET_CAST
+%token T_NEW
+%token T_CLONE
+%token T_EXIT
+%token T_IF
+%token T_ELSEIF
+%token T_ELSE
+%token T_ENDIF
+%token T_ECHO
+%token T_DO
+%token T_WHILE
+%token T_ENDWHILE
+%token T_FOR
+%token T_ENDFOR
+%token T_FOREACH
+%token T_ENDFOREACH
+%token T_DECLARE
+%token T_ENDDECLARE
+%token T_AS
+%token T_SWITCH
+%token T_ENDSWITCH
+%token T_CASE
+%token T_DEFAULT
+%token T_BREAK
+%token T_CONTINUE
+%token T_GOTO
+%token T_FUNCTION
+%token T_CONST
+%token T_RETURN
+%token T_TRY
+%token T_CATCH
+%token T_FINALLY
+%token T_THROW
+%token T_USE
+%token T_INSTEADOF
+%token T_GLOBAL
+%token T_STATIC
+%token T_ABSTRACT
+%token T_FINAL
+%token T_PRIVATE
+%token T_PROTECTED
+%token T_PUBLIC
+%token T_VAR
+%token T_UNSET
+%token T_ISSET
+%token T_EMPTY
+%token T_HALT_COMPILER
+%token T_CLASS
+%token T_TRAIT
+%token T_INTERFACE
+%token T_EXTENDS
+%token T_IMPLEMENTS
+%token T_OBJECT_OPERATOR
+%token T_DOUBLE_ARROW
+%token T_LIST
+%token T_ARRAY
+%token T_CALLABLE
+%token T_LINE
+%token T_FILE
+%token T_DIR
+%token T_CLASS_C
+%token T_TRAIT_C
+%token T_METHOD_C
+%token T_FUNC_C
+// %token T_COMMENT for now throw away in lexer
```

Comment:
Is this their comment or ours? It's phrased a bit oddly...

Reply:
That's ours. I wasn't sure if I should just delete the whole token, as we are currently not using it (we throw away comments in the lexer). On the other hand, we might want to use it in the future, e.g. for incremental parsing.

Reply:
Can we at least phrase it so that it's clear what is meant (I'm not really sure exactly what it means)?

Reply:
Okay, will rephrase. There are a few other commented out things, that we don't support, e.g. `T_INLINE_HTML`, whatever that is.

Reply:
I imagine that's inline HTML? Presumably you can just drop stuff between `<...>` tags?

Why don't we pull them all out to somewhere and say "here are tokens we don't support" or similar?